### PR TITLE
fix: use subtype=http and action=$method for all HTTP spans

### PR DESCRIPTION
**Pull request by trentm**
Use subtype 'http', even for https:// and http2 requests.

Fixes: #2075

### Checklist

- [x] Implement code
- [x] Add tests
- [x] Add CHANGELOG.asciidoc entry
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/master/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

**Comment by trentm** [comment]
I will add this CHANGELOG entry once the changelog updates from https://github.com/elastic/apm-agent-nodejs/pull/2108 are merged later today (hopefully):

```
* Fix inconsistencies in HTTP spans from other APM agents.
  <<span-subtype, `span.subtype`>> will now be "http" for HTTP, HTTPS, and
  HTTP/2 outgoing spans -- previously it was "http", "https", "http2",
  respectively. As well, <<span-action, `span.action`>> will now be the HTTP
  method (e.g. "GET", "PUT", "POST"), rather than "http". ({pull}2075[#2075])
```

File: lib/instrumentation/http-shared.js [hunk 1]
```diff
@@ -130,7 +130,7 @@ exports.traceOutgoingRequest = function (agent, moduleName, method) {
     return function (...args) {
       // TODO: See if we can delay the creation of span until the `response`
       // event is fired, while still having it have the correct stack trace
-      var span = agent.startSpan(null, 'external', moduleName, 'http')
+      var span = agent.startSpan(null, 'external', 'http')
```
Comment (astorm) [inline lib/instrumentation/http-shared.js:133]: nit: It looks like `span.action = req.method` could be removed by passing in req.method as the fourth argument here.

**Review by astorm — APPROVED** [review]
Approving -- looks like it does the thing.

File: lib/instrumentation/http-shared.js (hunk 1 above)
Comment (trentm) [reply lib/instrumentation/http-shared.js:133]: We don't have `req` until we do `      var req = orig.apply(this, newArgs)` a little while *after* the span is created.

**Pull request merged**